aibolit/patterns/non_final_class/non_final_class.py

```diff
@@ -11,9 +11,15 @@ def __init__(self):
         pass
 
     def value(self, filename: str) -> List[LineNumber]:
-        tree = build_ast(filename)
-        classes = tree.filter(ClassDeclaration)
-        return [
-            node.position.line for _, node in classes
-            if len([v for v in ['final', 'abstract'] if v in node.modifiers]) == 0
-        ]
+        tree = AST.build_from_javalang(build_ast(filename))
+        nodes = tree.nodes_by_type(ASTNodeType.CLASS_DECLARATION)
+
+        positions = []
+        for node in nodes:
+            line = tree.get_attr(node, 'source_code_line')
+            modifiers = tree.get_attr(node, 'modifiers')
```

Comment:
There is no such thing as `modifiers` yet. All those fields will be added soon, but now you have to extract them on your own. If you print AST, you will see that CLASS_DECLARATION node has child with type COLLECTION. There are some nodes with strings, that will represent modifiers. To access child with particular type, you could use `children_with_type` method of AST.